### PR TITLE
Add support for cleartype text antialiasing

### LINKED ISSUE #1298: Feature Request: Antialiasing mode options
# Summary of the new feature/enhancement

Support for different text rendering modes, current mode at least for me causes eye strain making new terminal unusable.
Example between terminal (left) and conemu (right):
text rendering terminal and conemu

# Proposed technical implementation details 
Adding options to `profiles.json`:
```
{
...
"TextAntialiasMode": "D2D1_TEXT_ANTIALIAS_MODE_CLEARTYPE",
...
}
```

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

I needed to do something to keep sane so today I day of learned about antialiasing. This PR adds the ability to specify the `"antialiasingMode"` as a setting.
* "antialiasingMode": "grayscale": the current behavior, `D2D1_TEXT_ANTIALIAS_MODE_GRAYSCALE`
* "antialiasingMode": "cleartype": use `D2D1_TEXT_ANTIALIAS_MODE_CLEARTYPE` instead

## PR Checklist
* [x] Closes #1298
* [x] I work here
* [ ] I didn't add tests 
* [x] Requires documentation to be updated

## Detailed Description of the Pull Request / Additional comments
Grayscale:
 
image

Cleartype:
image

 

Side-by-side (can you tell which is which?) 
 
image

### CODE REVIEW

**Comment by miniksa** [comment]
Please do put aliased in as an option.

**Comment by DHowett-MSFT** [comment]
Should this be the beginning of the `font` _object_? I am envisioning a future where we...

```
"font": {
    "family": "Cascadia Code",
    "weight": 400,
    "features": ["dlig", "ss01"]
}
```

but of course, maybe it chould also support a string (`"font": "Cascadia Code"`)

**Comment by zadjii-msft** [comment]
@DHowett-MSFT I'm gonna argue _no_, because I want this for 1.0 😄 


That's a fine idea for 1.x/backlog IMO. If "font" is a string, use that. If it's a object, try parsing other bits out of it. That'll work well enough, but won't land this side of 1.0

**Comment by zadjii-msft** [comment]
image

File: doc/cascadia/profiles.schema.json [hunk 1]
```diff
@@ -378,6 +378,16 @@
           "minimum": 0,
           "type": "number"
         },
+        "antialiasingMode": {
+          "default": "grayscale",
+          "description": "Controls how text is antialiased in the renderer. Possible values are \"grayscale\", \"cleartype\" and \"aliased\". Note that changing this setting will require starting a new terminal instance.",
```
Comment (miniksa) [inline doc/cascadia/profiles.schema.json]: Explain the default maybe? Grayscale is faster than cleartype but doesn't look as jagged as aliased. It's a compromise.

File: src/renderer/dx/DxRenderer.cpp [hunk 2]
```diff
@@ -2115,3 +2117,18 @@ void DxEngine::SetSelectionBackground(const COLORREF color) noexcept
                                         GetBValue(color) / 255.0f,
                                         0.5f);
 }
+
+// Routine Description:
+// - Changes the antialiasing mode of the renderer. This must be called before
+//   _PrepareRenderTarget, otherwise the renderer will default to
+//   D2D1_TEXT_ANTIALIAS_MODE_GRAYSCALE. This has only been tested with
```
Comment (DHowett-MSFT) [inline src/renderer/dx/DxRenderer.cpp]: it is no longer true that it has been tested with only those :smile: right?

Comment (zadjii-msft) [reply src/renderer/dx/DxRenderer.cpp]: oof okay fine

**Pull request merged**